newsroom/topics/topics_async.py

```diff
@@ -0,0 +1,243 @@
+
+
+class TopicResourceModel(NewshubResourceModel):
+    label: str
+    query: Optional[str] = None
+    filter: Optional[Dict[str, Any]] = None
+    created_filter: Annotated[Optional[Dict[str, Any]], Field(alias="created")] = None
+    user: Annotated[Optional[ObjectIdField], validate_data_relation_async("users")] = None
+    company: Annotated[Optional[ObjectIdField], validate_data_relation_async("companies")] = None
+    is_global: bool = False
+    subscribers: Optional[List[TopicSubscriber]] = []
+    timezone_offset: Optional[int] = None
+    topic_type: TopicType
+    navigation: Optional[List[Annotated[ObjectIdField, validate_data_relation_async("navigations")]]] = None
+    folder: Annotated[Optional[ObjectIdField], validate_data_relation_async("topic_folders")] = None
+    advanced: Optional[Dict[str, Any]] = None
+
+
+class TopicService(NewshubAsyncResourceService[TopicResourceModel]):
+    async def on_create(self, docs: List[TopicResourceModel]) -> None:
+        return await super().on_create(docs)
+
+    async def on_update(self, updates: Dict[str, Any], original: TopicResourceModel) -> None:
+        await super().on_update(updates, original)
+        # If ``is_global`` has been turned off, then remove all subscribers
+        # except for the owner of the Topic
+        if original.is_global and "is_global" in updates and not updates.get("is_global"):
+            # First find the subscriber entry for the original user
+            subscriber = next(
+                (
+                    subscriber
+                    for subscriber in (updates.get("subscribers") or original.subscribers or [])
+                    if subscriber.user_id == original.user
+                ),
+                None,
+            )
+
+            # Then construct new array with either subscriber found or empty list
+            updates["subscribers"] = [subscriber] if subscriber is not None else []
+
+        if updates.get("folder"):
+            updates["folder"] = ObjectId(updates["folder"])
+
+    async def on_updated(self, updates: Dict[str, Any], original: TopicResourceModel) -> None:
+        await super().on_updated(updates, original)
+        current_user = get_user()
+
+        if current_user:
+            await auto_enable_user_emails(updates, original, current_user)
+
+    async def on_delete(self, doc: TopicResourceModel):
+        await super().on_delete(doc)
+        users = await UsersService().search(lookup={"dashboards.topic_ids": doc.id})
+        async for user in users:
+            updates = {"dashboards": user.dashboards.copy()}
+            for dashboard in updates["dashboards"]:
+                dashboard["topic_ids"] = [topic_id for topic_id in dashboard["topic_ids"] if topic_id != doc.id]
+            await UsersService().update(user.id, updates)
+
+    async def on_user_deleted(self, sender, user, **kwargs):
+        # delete user private topics
+        await self.delete({"is_global": False, "user": user["_id"]})
```

Comment:
This does not match the expected signature of the `delete` method, as it expects a ResourceModel instance.
What you're after here is the newer [delete_many](https://github.com/superdesk/superdesk-core/blob/async/superdesk/core/resources/service.py#L412) method on the service